pkg/controller/storagealert/storagealert_controller.go

```diff
@@ -101,70 +102,26 @@ func (r *ReconcileStorageAlert) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	f := manifests.NewFactory(instance.Namespace)
-	prometheusK8sRules, err := f.PrometheusK8sRules()
-	prometheusK8sRules.Namespace = instance.Spec.StorageAlert.PrometheusNamespace
-
-	// Set StorageAlerts instance as the owner and controller
-	// if err := controllerutil.SetControllerReference(instance, prometheusK8sRules, r.scheme); err != nil {
-	// 	return reconcile.Result{}, err
-	// }
-
-	CreateOrUpdatePrometheusRule(prometheusK8sRules)
-
-	common.CreateOrUpdateRBAC()
-	err = common.CreateOrUpdateRBAC()
-	if err == nil {
-		reqLogger.Info("Created/Updated RBAC")
-	}
-
-	nsLabel := make(map[string]string)
-	nsLabel["openshift.io/cluster-monitoring"] = "true"
-	err = common.AddLabelToNamespace("openshift-storage", nsLabel)
-	if err == nil {
-		reqLogger.Info("Added label to Namespace", "Namepace :", "openshift-storage")
-	}
-
-	svcMonitorLabel := make(map[string]string)
-	svcMonitorLabel["app"] = "rook-ceph-mgr"
-	svcMonitorLabel["rook_cluster"] = "openshift-storage"
-	err = prometheus.CreateOrUpdateServiceMonitors("rook-ceph-mgr", "http-metrics", "openshift-storage", svcMonitorLabel)
-	if err == nil {
-		reqLogger.Info("Created/Updated ServiceMonitor", "ServiceMonitor :", "rook-ceph-mgr")
-	}
-
-	return reconcile.Result{}, nil
-}
-
-// CreateOrUpdatePrometheusRule Creates or Updates prometheusRule object
-func CreateOrUpdatePrometheusRule(p *monv1.PrometheusRule) {
-	reqLogger := log.WithValues("Prometheus Namespace: ", p.ObjectMeta.Namespace)
-	cfg, err := config.GetConfig()
-	mclient, err := monitoring.NewForConfig(cfg)
-	pclient := mclient.MonitoringV1().PrometheusRules(p.GetNamespace())
-	oldRule, err := pclient.Get(p.GetName(), metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		_, err := pclient.Create(p)
-		if err != nil {
-			reqLogger.Error(err, "creating PrometheusRule object failed", "Prometheus Namespace: ", p.ObjectMeta.Namespace)
-			return
-		}
-		reqLogger.Info("PrometheusRule Created.", "Prometheus Namespace: ", p.ObjectMeta.Namespace)
-		return
+	// Check prerequisites
+	err = tasks.Prerequisites(instance)
+	if err != nil {
+		// Prerequisites not met, requeue
+		reqLogger.Error(err, "Prerequisites not met")
```

Comment:
Broken sentence `Prerequisites not met` instead use `you have not met a Prerequisite`.

Reply:
ack.